### PR TITLE
First pass at reducer refactor

### PR DESCRIPTION
**Pull request by hudson-prestidge**
🍌I'm not super happy with how the 'player attack' case looks, but the reducer is way smaller/drier, the player movement action creator is smaller and also theoretically allows for moving multiple squares at once, and the tests are all passing 🍌

### CODE REVIEW

File: client/components/game.js [hunk 1]
```diff
@@ -33,17 +33,19 @@ class Game extends React.Component {
...
-    document.addEventListener("keyup", (evt) => {
+    document.addEventListener("keydown", (evt) => {
+        var {y, x} = player.position
+
       switch(evt.key) {
 
         case('ArrowLeft'):
           presentEnemy = enemies.find(function(enemy) {
-            return (enemy.position.y == player.position.y && enemy.position.x == player.position.x-1)
+            return (enemy.position.y == y && enemy.position.x == x-1)
```
Comment (madeleine-rashbrooke) [inline client/components/game.js:43]: this is a small-but-very-nice change - should it also be in the Arrow Right case on line 58? (I don't seem to be able to put a comment there as it hasn't changed) :+1:

File: client/components/game.js [hunk 2]
```diff
@@ -88,7 +90,7 @@ class Game extends React.Component {
           if(presentEnemy) {
             this.props.playerAttack(presentEnemy)
           }else{
-            this.props.playerMoveDown()
+            this.props.playerMove( y + 1, x )
```
Comment (madeleine-rashbrooke) [inline client/components/game.js:93]: Loving the shift to generic playerMove function :heart:

**Comment by madeleine-rashbrooke** [comment]
Hey this looks super awesome - very impressed by what you've achieved. Only one place where I think you've missed a small refactor to match the rest. :pineapple: :bangbang: :banana:

File: client/reducers/reducer.js [hunk 3]
```diff
@@ -6,8 +6,8 @@ function reducer (state = initialState, action) {
 
   var newState = Object.assign({}, state)
   var { tileGrid, enemies, player } = newState
-  var i = state.player.position.y
-  var j = state.player.position.x
+  var playerX = player.position.x
+  var playerY = player.position.y
```
Comment (DylanHul) [inline client/reducers/reducer.js:10]: 🍍

File: client/reducers/reducer.js [hunk 4]
```diff
@@ -20,61 +20,43 @@ function reducer (state = initialState, action) {
...
+          nextLevelFunc()
+        }
+        return newState
 
     //these are the cases for the player attacking
 
     case 'PLAYER_ATTACK':
-      var attackedEnemy = newState.enemies.find(function(enemy){
-        return enemy.position.x == action.payload.position.x && enemy.position.y == action.payload.position.y
+      var { loggedMessages } = newState
```
Comment (DylanHul) [inline client/reducers/reducer.js]: is this being used?

**Pull request merged**